### ISSUE TITLE
Recomendação de lib parallax

### ISSUE THREAD

**Issue by matheusggds** [frontendbr/forum#755]
Boa tarde Front-endeiros!!!!

Então, tenho uma demanda até segunda-feira - _nemqueriaferiadomesmo_ - para fazer uma landing page com MUITOS elementos no efeito parallax/floating on scroll. 

A refêrencia é esse site: http://www.loisjeans.com/web2012/es

As vezes que usei parallax foi mais em image background e coisas mais simples, alguma alma **linda e bondosa** já fez algo parecido e pode me indicar alguma lib? (pff).

Obrigado galera <3

**Comment by bernardodiasc** [comment]
[Antes de mais nada, @matheusggds o assunto é legal mas o detalhe do prazo que voce possui me deixou preocupado com a qualidade que voce vai conseguir entregar. Meus comentarios abaixo nao sao de maneira alguma uma critica direcionada a qualquer pessoa, porém muito direcionadas as más praticas que vemos no mercado...]

Domingos e feriados são sagrados. Acho sacanagem alguém pedir tanta fodacidade aos custos da vida e sanidade do desenvolvedor. Se todo projeto for dirigido de modo a explorar sem dó nem piedade ambos projetos open source (a comunidade) tanto como os desenvolvedores, esperando que uma lib pronta solucione instantaneamente os problemas do cliente, estaremos cavando a própria cova da nossa profissão.

Tenho lutado arduamente contra essa mentalidade, me desculpem. Parallax é um efeito muito antigo, onde ao movimentar a cena global, os elementos se movem individualmente em velocidades distintas, gerando assim uma sensação de profundidade. Essa técnica existe desde os primórdios das animações gráficas (lembra dos arcades? ja tinha isso tb), muitos projetos em Flash eram feitos dessa forma e atualmente a técnica é feita combinando CSS e JS.

De coração, querer fazer um site como esse da referência em poucos dias é absurdo. Parallax é um efeito complicado que requer dezenas de testes, e se suportar responsividade multiplica essas dezenas por mais meia dúzia. Se for suportar IE, mano, chora pq nem dormir voce vai poder nos próximos dias (considero inaceitável isso, qualquer patrão/gerente que te poe sobre essa pressão é um monstro e não sabe o que está fazendo)...

Enfim, se mesmo depois de eu argumentar contra a mentalidade de maximizar custos sacrificando os operários, voce ainda querer uma biblioteca open source pra resolver todos seus problemas sem que voce tenha que de fato aprender a fazer o efeito, acho essa lib http://mynameismatthieu.com/WOW/ muito boa. Manera nos efeitos pois alcançar baranguiçe extrema com isso é bem fácil.

Agora, realmente gostaria de recomendar que tirasse o tempo necessário (nada de deadline pra ontem pq isso é muito amadorismo) para aprender a escrever efeitos de parallax. 
É muito interessante e vale a pena aprender.

Colocando a parte tecnica para fazer o efeito de um modo bem simples:

- use JS pra observar o evento de scroll e checar a visibilidade no viewport dos elementos
- use CSS para definir os estados visuais dos elementos, ou seja, por exemplo, estado inicial e estado final. O JS altera as classes dos elementos e o então CSS faz a animação de transição 
- use data-attr no HTML pra ficar bem facil de configurar os elementos que voce quer com efeito e como voce quer que os efeitos sejam em particular (velocidade, delay, etc)

Tem outras libs uteis pra isso, não hesite em ler o código fonte delas. :)

**Comment by LFeh** [comment]
@bernardodiasc Sem muito o que complementar na sua resposta, sensacional.

E digo mais, observem que o site do exemplo em questão foi produzido em 2012 e hoje, em 2017, continua LENTO. 

Parallax é um efeito complexo, cheio de problemas de compatibilidade e performance, que desencorajo fortemente designers a usarem. Como colocado pelo @bernardodiasc são necessários muitos testes e otimizações para chegar em um resultado legal. 

Em todo caso...

Das vezes que precisei desenvolver um (sim, não escapei do hype de 2012 e 2013 dessa tecnica), gostava do [ScrollMagic](http://scrollmagic.io/).

**Comment by matheusggds** [comment]
@bernardodiasc @LFeh 

Concordo com você em praticamente tudo, e eu tento acompanhar/ajudar onde consigo e posso a comunidade... infelizmente, muitas vezes irão ocorrer demandas desse tipo. Obviamente já alertei meus supervisores que é difícil ficar com essa qualidade em 4 dias, sozinho impossível. Não precisa ter versão mobile, o que ajuda e MUUUUITO meu trabalho, eu entendo, compreendo e concordo em partes com sua opnião @bernardodiasc , mas acredito que algumas coisas nunca mudarão, como demandas corridas e urgentes, o mais importante é você ter consciência de custo/benefício na sua vida profissional, se você quer se doar pela empresa, se a empresa merecesse esse seu esforço, e dai você decidir suas atitudes.

Obrigado pelas indicações, de verdade, eu já as conhecia, mas eu adoro saber a opinião e visão dos outros mesmo que eu já tenha um conhecimento, sempre há alguma coisa nova a absorver.

Abraço, muito obrigado.

**Comment by willianjusten** [comment]
Concordo plenamente com o que o @bernardodiasc disse, infelizmente ainda existem essas demandas "malucas", mas precisamos sempre tentar não ceder a elas e explicar nossas razões para tal...

Sobre o que você deseja, assim como o @LFeh , aconselho a [ScrollMagic](http://scrollmagic.io/) que definitivamente é a melhor para tal e junto a ela, você pode acoplar o [GSAP](https://greensock.com/gsap) que é a melhor lib para qualquer tipo de animação.

Inclusive no [I hate tomatos](https://ihatetomatoes.net/a-simple-parallax-template-using-scrollmagic-and-greensock/) tem uns tutoriais mostrando o uso, que podem te ajudar.

**Comment by matheusggds** [comment]
Obrigado @willianjusten ! Vou dar uma olhada
